### PR TITLE
docs: add merge insert, create index and create scalar index to public rest api doc

### PR DESCRIPTION
**Pull request by eddyxu**
Added 3 APIs doc publicly:
- `merge_insert`
- `create_index`
- `create_scalar_index`

### CODE REVIEW

File: docs/openapi.yml [hunk 1]
```diff
@@ -219,20 +219,87 @@ paths:
...
+        - Data
+      summary: Merge Insert
+      operationId: mergeInsert
+      parameters:
+        - $ref: "#/components/parameters/table_name"
+        - name: on
+          in: query
+          description: |
+            The column to use as the primary key for the merge operation.
+          required: false
```
Comment (westonpace) [inline docs/openapi.yml:255]: I believe this is required (or should be)

File: docs/openapi.yml [hunk 2]
```diff
@@ -219,20 +219,87 @@ paths:
...
+            Rows that exist in both the source table (new data) and
+            the target table (old data) will be updated, replacing
+            the old row with the corresponding matching row.
+          required: false
+          schema:
+            type: boolean
+        - name: when_matched_update_all_filt
+          in: query
+          description: |
+            The filter expression that specifies the rows to update.
```
Comment (westonpace) [inline docs/openapi.yml:270]:
```suggestion
            If present then only rows that satisfy the filter expression will
            be updated
```

**Review by westonpace — APPROVED** [review]
I didn't check the actual phalanx endpoints but these descriptions seem correct.

File: docs/openapi.yml (hunk 1 above)
Comment (eddyxu) [reply docs/openapi.yml]: Fixed.

**Pull request merged**